promql/generated_parser.y.go

```diff
@@ -981,7 +981,7 @@ yydefault:
 			var err error
 			yyVAL.uint, err = strconv.ParseUint(yyDollar[1].item.Val, 10, 64)
 			if err != nil {
-				yylex.(*parser).errorf("invalid repitition in series values: %s", err)
+				yylex.(*parser).errorf("invalid repetition in series values: %s", err)
```

Comment:
This looks suspiciously like generated code/a third party. Let me know the origin and I'll see about scanning it instead…

Reply:
This is a generated file and should not be changed. The method of generation is in the first line of this file, which is `goyacc`.

Reply:
https://gitlab.com/cznic/goyacc/blob/master/ ? -- I'll have to look...

Reply:
Nope, golang.org/x/tools/cmd/goyacc .

The file is generated from `promql/generated_parser.y`.

We probably should document this somewhere.

Reply:
Err, so I should have kept the change https://github.com/prometheus/prometheus/commit/6a6d61f6439e40a109eff25b741bf25d730560fa to: https://github.com/prometheus/prometheus/blob/213a8fe89a7308e73f22888a963cbf9375217cd6/promql/generated_parser.y#L348 ?


(That filename is fairly confusing. I'm going to let the dust settle on this PR and then I can go back and touch it. -- And yes, I understand the tradition of naming a `.y` file and a `.something` file with the same prefix.)

Reply:
> Err, so I should have kept the change [6a6d61f](https://github.com/prometheus/prometheus/commit/6a6d61f6439e40a109eff25b741bf25d730560fa) to:
> 
> https://github.com/prometheus/prometheus/blob/213a8fe89a7308e73f22888a963cbf9375217cd6/promql/generated_parser.y#L348
> ?
Yes, and then regenerate the file.

> (That filename is fairly confusing. I'm going to let the dust settle on this PR and then I can go back and touch it. -- And yes, I understand the tradition of naming a `.y` file and a `.something` file with the same prefix.)

I'm doing a lot of changes in that file at the moment anyway, so I can just fix that typo on the way.

And yes, the naming and generation process should be made more clear.

Reply:
This spelling error gets fixed as a side effect of #6548.